### ISSUE TITLE
Add option for `--prefix` and `--root`

### ISSUE THREAD

**Issue by naveen521kk** [FFY00/python-install#8]
I want the `--prefix` and `--root` arguments to work as it is done on setuptools/distutils (when using `setup.py install`) or pip's. 

I was trying to use this on MSYS2's packages but I couldn't find an option to change the prefix without which it would use absolute windows path which can't be used in packages. For example, see file difference on a CI run [here](https://github.com/msys2/MINGW-packages/runs/4752543654?check_suite_focus=true#step:10:736).

Thanks for this project.

**Comment by FFY00** [comment]
We have `--destdir`.

**Comment by naveen521kk** [comment]
> We have --destdir.

Yeah but still I won't be able to change what will be the prefix of the install. For example, in the above linked CI log, the file gets installed to `$pkgdir/M/msys64/mingw64/lib/python3.9/site-packages/typing_extensions.py` while it should be `/mingw64/lib/python3.9/site-packages/typing_extensions.py`. If I pass `/mingw64` as a prefix in which case I should be able to the required output.

**Comment by FFY00** [comment]
I understand, and while that was an approach that worked, though I wouldn't recommend, in the past, with Python 3.10 it no longer becomes viable. In fact, it was never viable with downstreams like Debian patching Python install locations (see https://ffy00.github.io/blog/02-python-debian-and-the-install-locations/ for an in-depth writeup on some of that).

The question here is why isn't the Python interpreter you are using returning the correct path. Are you targeting a different interpreter, perhaps?

I know this is annoying, but IMO adding a feature to override the prefix like that will result in long term pain.

**Comment by naveen521kk** [comment]
> The question here is why isn't the Python interpreter you are using returning the correct path. 

Well, technically it's returning the correct path. MinGW python doesn't use POSIX emulation from Cygwin, so `/` wouldn't be the root of the MSYS2 installation, instead would be `C:\`. So it can't/wont return the correct installation path as we need. Hence, we have to pass the prefix manually so that it's relocatable and we can distribute it as a Pacman package.

I think without support for `--prefix`, it's not possible to use this project in MSYS2 MinGW based environments for packaging.
